### PR TITLE
[SuiteSparse] Attempt on updating and adding GPU support

### PR DESCRIPTION
**Pull request by ViralBShah**
Update to 5.8.1 and add SLIP_LU. 
Hoping to enable GPU support for CHOLMOD and SPQR.
Also hoping to enable Mongoose.

### CODE REVIEW

**Comment by ViralBShah** [comment]
Can't find `<gmp.h>`. @giordano Do you know if any special things are needed to be able to find the GMP header files?

File: S/SuiteSparse/SuiteSparse_GPU/build_tarballs.jl [hunk 1]
```diff
@@ -1,19 +1,19 @@
-using BinaryBuilder
+using BinaryBuilder, Pkg
 
 name = "SuiteSparse"
```
Comment (giordano) [inline S/SuiteSparse/SuiteSparse_GPU/build_tarballs.jl:3]: I'm a bit confused by this PR: did you _move_ `SuiteSparse`?

**Comment by giordano** [comment]
> Can't find <gmp.h>. @giordano Do you know if any special things are needed to be able to find the GMP header files?

Did you figure it out?  I don't see that error anywhere

**Comment by staticfloat** [comment]
I'm pretty sure we don't want to move `SuiteSparse`, but instead just copy it, and also it should be noted that since we already ship SuiteSparse with Julia you're going to have to build a special Julia that uses this package instead of the normal SuiteSparse.

**Comment by ViralBShah** [comment]
@staticfloat I was hoping that I could change the name, and then re-enable METIS, and build a bunch of other experimental stuff. And then have a separate package that doesn't clash with Base. Is that not ok?

@giordano gmp.h error here (I haven't figured it out): https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=7370&view=logs&j=dde15f49-fafd-5c47-4a6b-29ba8c8f6c9b&t=ed7f4e6b-29fd-5815-ef44-99325a0de85d&l=1830

**Comment by ViralBShah** [comment]
Do we depend upon this build of SuiteSparse in Base Julia? So maybe I want to leave the old one as is, and create a new SuiteSparse_GPU?

**Comment by giordano** [comment]
That's an old build, the latest one is https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=7371&view=logs&j=dde15f49-fafd-5c47-4a6b-29ba8c8f6c9b&t=ed7f4e6b-29fd-5815-ef44-99325a0de85d and compilation worked (upload failed because of the wrong `name` variable)

**Pull request closed without merging**